src/Microsoft.OData.Client/Materialization/FeedAndEntryMaterializerAdapter.cs

```diff
@@ -296,6 +323,83 @@ private ODataResourceSet ReadFeedCore(bool lazy)
             return result;
         }
 
+        /// <summary>
+        /// Reads an <see cref="ODataDeltaResourceSet"/>.
+        /// </summary>
+        /// <returns>The <see cref="MaterializerDeltaFeed"/> of the read <see cref="ODataDeltaResourceSet"/>.</returns>
+        private MaterializerDeltaFeed ReadDeltaFeedCore()
+        {
+            this.ExpectState(ODataReaderState.DeltaResourceSetStart);
+
+            ODataDeltaResourceSet result = (ODataDeltaResourceSet)this.reader.Item;
+            MaterializerDeltaFeed feed = null;
+
+            if (result == null)
+            {
+                return null;
+            }
+
+            feed = MaterializerDeltaFeed.CreateDeltaFeed(
+                    result,
+                    new List<IMaterializerState>(),
+                    this.materializerContext);
+
+            if (this.items?.Count > 0)
+            {
+                this.items.Push(feed);
+            }
+            else
+            {
+                IMaterializerState topLevelItem = feed;

```

Comment:
1. Given that a `MaterializerDeltaFeed` is an `IMaterializerState`, is this line necessary?
2. The `if` blocks checks that there are items in the stack before pushing `feed` into the stack. The `else` block just stashes the same `feed` into the stack. What's the difference between what is happening in the `if` block as opposed to what is happening in the `else` block? Why check that `this.items?.Count > 0` in the `if` block if you it's possible to have `feed` as a top-level item?

Reply:
Yes..I think we do not need the `if --else--`